packages/dds/tree/src/simple-tree/tree.ts

```diff
@@ -84,11 +87,29 @@ export interface ITreeConfigurationOptions {
 	 * bit slower.
 	 */
 	enableSchemaValidation?: boolean;
+
+	/**
+	 * If `true`, constructing the configuration will error if any valid input data could ambiguously parse in multiple ways.
+	 *
+	 * @defaultValue `false`.
+	 *
+	 * @remarks
+	 * When this is true, it ensures that the compile time type safety for data when constructing nodes is sufficient to ensure than the runtime behavior will node give data ambiguity errors.
+	 *
+	 * This ensures that the canonical JSON representation of the node (the default produce by JSON.stringify before any customization) of any union in the tree is lossless and unambiguous.
+	 *
+	 * This check is conservative: some complex cases may error if the current simple algorithm can not show no ambiguity is possible.
+	 * This check may become more permissive over time.
+	 * @privateRemarks
+	 * In the future (mostly by making map node allow record like inputs), we can support lossless round tripping via tha canonical JSON representation mentioned above when unambiguous.
+	 */
+	readonly ensureUnambiguous?: boolean;
```

Comment:
This option is one of the more important parts of this proposal.

Later we can add "constant" fields to allow more disambiguation.

Reply:
Can you elaborate on what you mean by "constant" fields here?

Reply:
The idea is to allow defining fields on object nodes which are not persisted, and always have a constant value. So you could make a field on Foo that's always `type: "Foo"` and on Bar thats always `type: "Bar"`, which would disambiguate a union of FOO | Bar, even if all their other fields are optional. This constant fields could be included on the TreeNode objects and the json versions (unless you use the stable keys I guess, but we don't support round tripping those unless use use the verbose format which doesn't need the constant fields)

Reply:
Makes sense. Thanks for the clarification!